proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java

```diff
@@ -596,6 +601,28 @@ public void handle(Commands commands) throws Exception
         }
     }
 
+    @Override
+    public void handle(final Chat chat) throws Exception
+    {
+        ChatMessageType position = ChatMessageType.values()[chat.getPosition()];
+        Supplier<BaseComponent[]> msgSupplier = new Supplier<BaseComponent[]>() {
+            @Override
+            public BaseComponent[] get() {
+                return ComponentSerializer.parse( chat.getMessage() );
+            }
+        };
+        ServerChatEvent chatEvent = new ServerChatEvent( con.getServer(), con, msgSupplier, position );
+
+        if ( !bungee.getPluginManager().callEvent( chatEvent ).isCancelled() )
+        {
+            // Be aware that because of this, all server-sent chat messages are parsed and re-serialised
```

Comment:
A lot of the other events in this situation check to see if any plugins changed the result, otherwise they just forward the packet.
Maybe you could do that here? Although that would then negate laziness (cost a json decode), but save a json encode.....
What's to be aware? I think you mentioned a bug?

Reply:
The encode should be only be needed if:

- the getter was called or
- the setter was called.

Reply:
PTAL at 8e18a5c and see if that addresses this